dm/pkg/etcdutil/etcdutil_test.go

```diff
@@ -174,26 +177,28 @@ func (t *testEtcdUtilSuite) testMemberUtilInternal(c *C, portCount int) {
 }
 
 func (t *testEtcdUtilSuite) testRemoveMember(c *C) {
-	cluster := integration.NewClusterV3(tt, &integration.ClusterConfig{Size: 3})
-	defer cluster.Terminate(tt)
-
-	i := cluster.WaitLeader(tt)
-	c.Assert(i, Not(Equals), -1)
+	cluster := integration.NewClusterV3(t.testT, &integration.ClusterConfig{Size: 3})
+	defer cluster.Terminate(t.testT)
 
-	cli := cluster.Client(i) // client to the leader.
+	leaderIdx := cluster.WaitLeader(t.testT)
+	c.Assert(leaderIdx, Not(Equals), -1)
+	cli := cluster.Client(leaderIdx) // client to the leader.
 
 	respList, err := ListMembers(cli)
 	c.Assert(err, IsNil)
 	c.Assert(respList.Members, HasLen, 3)
 
 	// always try to remove the first member, may or may not the leader
 	respRemove, err := RemoveMember(cli, respList.Members[0].ID)
-	c.Assert(err, IsNil)
-	c.Assert(respRemove.Members, HasLen, 2)
-
-	// get another client
-	cli = cluster.RandClient()
+	if err != nil {
+		// remove the leader will meet this error, and next line will check the member count
+		c.Assert(err, ErrorMatches, ".*etcdserver: server stopped")
+	} else {
+		c.Assert(respRemove.Members, HasLen, 2)
+	}
```

Comment:
can split this to two situation, for leader and non-leader

Reply:
after split to two test case , i found that it's nothing about leader, it's about if we use a client that connected to the member we want to remove we **may** meet `etcdserver: server stopped` this error. so this is a unsteable feautre in etcd self see more in https://github.com/etcd-io/etcd/pull/7242

fixed in https://github.com/pingcap/tiflow/pull/4175/commits/3a71b1270dfdbba84e35519d70ed8f982fd17906